test/run

```diff
@@ -143,24 +161,30 @@ def print_counter(counter):
 
 
 def main():
+    args = get_args()
     logging.basicConfig(format="run-tests: %(message)s")
     test_dir = pathlib.Path(__file__).resolve().parent
     root = test_dir.parent
     litani = root / "litani"
+    if args.output_dir:
+        output_dir = pathlib.Path(args.output_dir)
+    else:
+        output_dir = test_dir / "output"
 
     run_cmd([
         litani, "init",
         "--project", "Litani Test Suite",
-        "--output-prefix", test_dir / "output",
-        "--output-symlink", test_dir / "output" / "latest"])
+        "--output-prefix", output_dir,
+        "--output-symlink", output_dir / "latest"])
 
     counter = {
         "added": 0,
         "total": 0,
     }
 
     add_unit_tests(litani, test_dir, root, counter)
-    add_e2e_tests(litani, test_dir, root, counter)
+    add_e2e_tests(
+        litani, test_dir, counter, output_dir=args.output_dir or None)
```

Comment:
Why not use the value of `output_dir`?

Reply:
Because in `add_e2e_tests()` there is no use of `test_dir / "output"` which is the alternate value when `args.output_dir` is not passed. Reason it is not used is because the default run dir for e2e tests is created under `test_dir / "e2e" / "output`

Reply:
If I were the user and I specified `--output_dir <my-directory>`, I would expect the behavior to be either:
* that the tests that currently end up in `test_dir / "output"` to end up in `<my-directory> / "output"` and the tests that end up in `test_dir / "e2e" / "output"` to end up in `<my-directory> / "e2e" / "output"`

or
* that the tests that currently end up in `test_dir / "output"` to end up in `<my-directory>` and the tests that end up in `test_dir / "e2e" / "output"` to end up in `<my-directory> / "e2e" / "output"`

Reply:
The first one is fine. The most important thing about this PR is that when `--output-dir` is passed, there should be no files written outside that directory. That includes not only the top-level Litani test process, but also all of the Litani invocations that are part of each e2e test.

The reason for this PR is that if we install Litani in a root-owned location, we want to be able to run the test suite and guarantee that it won't try to write output files under the root-owned location where Litani is installed. So if we run the test suite with `--output-directory /tmp/litani`, absolutely everything needs to be written under there.

Probably the easiest way to try this out is just to try it, `sudo chown -R root: litani_checkout` and then try running the test suite (without sudo) and seeing if anything breaks.